source/Engine.cpp

```diff
@@ -1414,7 +1414,7 @@ void Engine::CalculateStep()
 
 	// Step the weather.
 	for(Weather &weather : activeWeather)
-		weather.Step(newVisuals);
+		weather.Step(newVisuals, flagship ? flagship->Position() : Point());
```

Comment:
```suggestion
		weather.Step(newVisuals, center);
```
If the flagship dies then the hazard generates at the wrong position.

Reply:
Thought so.